store/sqlstore/post_store.go

```diff
@@ -647,35 +648,19 @@ func (s *SqlPostStore) GetPostsSince(options model.GetPostsSinceOptions, allowFr
 			) j ON p1.Id = j.Id
           ORDER BY CreateAt DESC`
 	} else if s.DriverName() == model.DATABASE_DRIVER_POSTGRES {
-		query = `
-			(SELECT
-                       *` + replyCountQuery1 + `
-               FROM
-                       Posts p1
-               WHERE
-                       (UpdateAt > :Time
-                               AND ChannelId = :ChannelId)
-                       LIMIT 1000)
-               UNION
-                       (SELECT
-                           *` + replyCountQuery2 + `
-                       FROM
-                           Posts p2
-                       WHERE
-                           Id
-                       IN
-                           (SELECT * FROM (SELECT
-                               RootId
-                           FROM
-                               Posts
-                           WHERE
-                               UpdateAt > :Time
-                                               AND ChannelId = :ChannelId
-                               LIMIT 1000) temp_tab))
-               ORDER BY CreateAt DESC`
+		query = `WITH cte AS (SELECT
+		       *
+		FROM
+		       Posts
+		WHERE
+		       UpdateAt > :Time AND ChannelId = :ChannelId
+		       LIMIT 1000)
```

Comment:
Does there need to be an ORDER BY in here to ensure deterministic results?

Reply:
The `ORDER BY` is done later after the UNION is done. If you recall https://community-daily.mattermost.com/core/channels/developers-server/s1ab7695z3rqumsgsdsj91gp1h, this API is known to be buggy and we just documented it as magic.

Reply:
Ah yeah, of course. We can open that can of worms another day (hopefully when this is handled by a different endpoint).